snmp/datadog_checks/snmp/snmp.py

```diff
@@ -363,7 +363,9 @@ def _check_with_config(self, config):
                 self.log.debug('Querying device %s', config.ip_address)
                 config.add_uptime_metric()
                 results, error = self.fetch_results(config, config.all_oids, config.bulk_oids)
-                self.report_metrics(config.parsed_metrics, results, config.tags)
+                tags = self.extract_metric_tags(config.parsed_metric_tags, results)
+                tags.extend(config.tags)
+                self.report_metrics(config.parsed_metrics, results, tags)
```

Comment:
Should this be be rewritten as…?

```python
metric_tags = self.extract_metric_tags(config.parsed_metric_tags, results)
metric_tags.extend(tags)  # 'tags' being assigned above
self.report_metric(config.parsed_metrics, results, metric_tags)
```

This is the same behavior but we reuse the list of `tags` defined on L355 now, and used in service checks, making it clearer that these are "shared" tags.

Reply:
I want them being used for service checks though? Not sure how to achieve that with your solution.

Reply:
Oh OK I see, yeah I didn't understand it that way, sounds good!